pkg/component/runtime/manager.go

```diff
@@ -658,6 +658,16 @@ func (m *Manager) CheckinV2(server proto.ElasticAgent_CheckinV2Server) error {
 		return status.Error(codes.PermissionDenied, "invalid token")
 	}
 
+	// enable chunking with the communicator if the initial checkin
+	// states that it supports chunking
+	runtime.comm.chunkingAllowed = false
+	for _, support := range initCheckin.Supports {
+		if support == proto.ConnectionSupports_CheckinChunking {
+			runtime.comm.chunkingAllowed = true
```

Comment:
Can we get this to be included in the `state.yaml` or some other part of the diagnostics so we can see if it is enabled or not when debugging?

Reply:
I will look at how difficult that is to propagate to get that information exposed.

Reply:
How much do we really want to propagate this information? At the moment this runtime communication is separate from the component state and no lock is required to ensure its consistent. I have looked into this and its rather difficult to propagate this information in a manner that it can be exposed in the diagnostics output.

Seems like a large amount of work for some information of a true/false where it should basically always become the default as the beats/agent versions are fixed to the same release cycle.

Reply:
I agree it is not worth introducing complexity or new locks for.

I still think we want a way to tell when this functionality is taking effect, my primary concern is not having a way to know it is causing us problems. Perhaps some well placed log lines can do this job effectively.

If we get a configuration flag to disable the chunking that gives us one way to isolate it as the cause of a problem, but without some observability into if it's actually happening we will still be guessing as to when to try that.

Reply:
I added log line for when it is enabled as well as a specific log line when it could be enabled, but it is explicitly disabled.